### ISSUE TITLE
TLS configuration via SNI is not working

### ISSUE THREAD

**Issue by korrat** [viridIT/vSMTP#1249]
**Describe the bug**
I am trying to configure a domain following the Doe family example using [SNI](https://vsmtp.rs/tuto/0/ssl-tls.html#certificate--sni). But somehow, I am unable to get vSMTP to accept my TLS certificate. Whenever I try to start the server, I get the following error: 

```
ERROR Receiver build failure. error=cannot add sni to resolver 'korr.at': unexpected error: The server certificate is not valid for the given name
```

When using the root config, the configuration is accepted without issue.

**Expected behavior**
vSMTP should accept the certificate in either configuration.

**System (please complete the following information):**
- vSMTP version:
  ```
  vsmtp v2.2.1                                                                                                                                                                                                                                                                                         
  commit: 5ab457489c292f36b646cd3a3cad54da4de4cd4d 
  ```
- OS: Linux 5.10.0-23-amd64 #1 SMP Debian 5.10.179-1 (2023-05-12) x86_64 GNU/Linux
- OS version: Debian 11

**Comment by korrat** [comment]
No, the certificate is for `mail.korr.at`, the name in the MX record for `korr.at`. Is this use case not supported?

**Comment by ltabis** [comment]
We just released the first release candidate for vSMTP v3.0 and TLS is not yet available for this version, so I am closing this.
